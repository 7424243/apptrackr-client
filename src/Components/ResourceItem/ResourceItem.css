.resource_item_content {
    margin-right: 25px;
    line-height: 2.5;
    font-size: larger;
}