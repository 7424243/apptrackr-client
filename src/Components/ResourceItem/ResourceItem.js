import React from 'react'
import './ResourceItem.css'
import PropTypes from 'prop-types'

function ResourceItem(props) {
    return (
        <li className='resource_item_content' key={props.id}>
            <a 
                href={props.url} 
                target='_blank' 
                rel='noreferrer'
            >{props.name} â‡¢
            </a>
        </li>
    )
}

export default ResourceItem

ResourceItem.propTypes = {
    id: PropTypes.number.isRequired,
    url: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired
}
