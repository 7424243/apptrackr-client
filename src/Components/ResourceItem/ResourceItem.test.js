import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import ResourceItem from './ResourceItem'

describe('ResourceItem Component', () => {

    const props = {
        url: 'https://www.google.com',
        name: 'test',
        id: 1
    }

    //smoke test
    it('renders without crashing', () => {
        const div = document.createElement('div')
        ReactDOM.render(<ResourceItem {...props}/>, div)
        ReactDOM.unmountComponentAtNode(div)
    })

    //snapshot test
    it('renders the UI as expected', () => {
        const tree = renderer
            .create(<ResourceItem {...props}/>)
            .toJSON()
        expect(tree).toMatchSnapshot()
    })
    
})