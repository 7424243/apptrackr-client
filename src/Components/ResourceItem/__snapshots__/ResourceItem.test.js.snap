// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResourceItem Component renders the UI as expected 1`] = `
<li
  className="resource_item_content"
>
  <a
    href="https://www.google.com"
    rel="noreferrer"
    target="_blank"
  >
    test
     â‡¢
  </a>
</li>
`;
